core/src/main/java/org/pentaho/di/connections/vfs/VFSConnectionDetails.java

```diff
@@ -34,4 +34,19 @@ public interface VFSConnectionDetails extends ConnectionDetails {
   default String getDomain() {
     return "";
   }
+
+  default boolean isSupportsRootPath() {
```

Comment:
Java doc comments for all all these functions and/or variables.

- VFSConnectionDetails#isSupportsRootPath
- VFSConnectionDetails#isRootPathRequired()
- VFSConnectionDetails#getRootPath()
- VFSConnectionDetails#setRootPath()


Assume developers not familiar with your work will come across these functions. They shouldn't have to dig through multiple projects to get the high level meaning of these functions/variables.